WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/RefundShippingLabelViewController.swift

```diff
@@ -0,0 +1,204 @@
+            self.showRefundResultNotice(result: result)
+        }
+        ServiceLocator.stores.dispatch(action)
+        onDismiss()
+    }
+
+    func showRefundResultNotice(result: Result<Yosemite.ShippingLabelRefund, Error>) {
+        let notice: Notice
+        switch result {
+        case .success:
+            let title = String.localizedStringWithFormat(Localization.refundSuccessNoticeFormat, shippingLabel.serviceName, viewModel.refundableAmount)
+            notice = Notice(title: title, feedbackType: .success)
+        case .failure(let error):
+            DDLogError("⛔️ Failed to request a refund for shipping label \(shippingLabel.shippingLabelID): \(error)")
+            notice = Notice(title: Localization.refundErrorNotice, feedbackType: .error)
+        }
+        noticePresenter.enqueue(notice: notice)
+    }
+}
+
+// MARK: Configuration
+private extension RefundShippingLabelViewController {
+    func configureNavigationBar() {
+        navigationItem.title = Localization.navigationBarTitle
+    }
+
+    func configureTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        registerTableViewCellsAndHeader()
+
+        tableView.tableFooterView = UIView()
+        tableView.backgroundColor = .basicBackground
+        tableView.estimatedSectionHeaderHeight = Constants.sectionHeight
+        tableView.sectionHeaderHeight = UITableView.automaticDimension
+        tableView.bounces = false
+    }
+
+    func registerTableViewCellsAndHeader() {
+        // Rows.
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+        // Header.
+        tableView.register(PlainTextSectionHeaderView.self, forHeaderFooterViewReuseIdentifier: PlainTextSectionHeaderView.reuseIdentifier)
+    }
+
+    func configureRefundButton() {
+        refundButton.applyPrimaryButtonStyle()
+        refundButton.setTitle(viewModel.refundButtonTitle, for: .normal)
+        refundButton.on(.touchUpInside) { [weak self] _ in
+            self?.refundShippingLabel()
+        }
+    }
+
+    func configureCellCommonProperties(_ cell: ValueOneTableViewCell) {
+        cell.detailTextLabel?.textColor = Constants.cellValueTextColor
+        cell.selectionStyle = .none
+        cell.accessoryType = .none
+    }
+}
+
+extension RefundShippingLabelViewController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = rowAtIndexPath(indexPath)
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row)
+        return cell
+    }
+
+    private func rowAtIndexPath(_ indexPath: IndexPath) -> Row {
```

Comment:
Do you have any plans to reuse this function elsewhere? If not I don't see a problem with having this statement inline on `cellForRowAtIndexPath`, what do you think?

Reply:
yea there is really no benefit having it in a separate function, updated in https://github.com/woocommerce/woocommerce-ios/pull/3306/commits/9bd8ddd9b7ba9447f064a10049b4bca1cdde5a17